modules/state_machine/states/dash.gd

```diff
@@ -1,25 +1,31 @@
 extends "state.gd"
 var init_pos = Vector2()
+signal drifting
+signal dashing(speed)
```

Comment:
Maybe move this `dashing` behavior to the jump setup, together with other signals only being used for modifying the `jump.in_jump_speed`